contribs/github-bot/internal/requirements/reviewer.go

```diff
@@ -11,6 +11,38 @@ import (
 	"github.com/xlab/treeprint"
 )
 
+// deduplicateReviews returns a list of reviews with at most 1 review per
+// author, where approval/changes requested reviews are preferred over comments
+// and later reviews are preferred over earlier ones.
+func deduplicateReviews(reviews []*github.PullRequestReview) []*github.PullRequestReview {
+	added := make(map[string]int)
+	result := make([]*github.PullRequestReview, 0, len(reviews))
+	for _, rev := range reviews {
+		idx, ok := added[rev.User.GetLogin()]
+		switch utils.ReviewState(rev.GetState()) {
+		case utils.ReviewStateApproved, utils.ReviewStateChangesRequested:
+			// this review changes the "approval state", and is more relevant,
+			// so substitute it with the previous one if it exists.
+			if ok {
+				result[idx] = rev
+			} else {
+				result = append(result, rev)
+				added[rev.User.GetLogin()] = len(result) - 1
+			}
```

Comment:
Since this block is meant to overwrite the review state with the most recent one for a given user, are we sure that the PRs will always be listed from the oldest to the newest? Or do we need to check the review timestamp to ensure this?

Reply:
Alright, forget it, I just got confirmation in the docs and we're good! 👍 

> The list of reviews returns in chronological order. 

https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28#list-reviews-for-a-pull-request